### PR TITLE
Update Transfer Model

### PR DESCRIPTION
**Pull request by gkmk**
Update Transfer model to support dynamic loading of overwritten Wallet or Transaction model classes. 

Defaults to current Wallet and Transaction model from inside the package if no overwritten models specified.

### CODE REVIEW

**Review by rez1dent3 — COMMENTED** [review]
Regarding the configuration, I don't mind, you're right. But there is no need to return morphTo. To_type & from_type will be removed in version 11.x.

**Comment by rez1dent3** [comment]
@gkmk Also, I ask you to write a unit test for this code, if it’s not difficult for you.

**Comment by rez1dent3** [comment]
@gkmk I looked at your code carefully and realized that this code is not necessary.
I added the following a long time ago and in the connections you should get the correct model.

```php
        $this->app->bind(Transaction::class, $configure['transaction']['model'] ?? null);
        $this->app->bind(Transfer::class, $configure['transfer']['model'] ?? null);
        $this->app->bind(Wallet::class, $configure['wallet']['model'] ?? null);
```

See: https://github.com/bavix/laravel-wallet/pull/859

File: src/Models/Transfer.php [hunk 1]
```diff
@@ -79,32 +80,32 @@ public function getTable(): string
...
+        return $this->morphTo();
     }
 
     /**
      * @return BelongsTo<Transaction, self>
      */
     public function deposit(): BelongsTo
     {
-        return $this->belongsTo(Transaction::class, 'deposit_id');
+        return $this->belongsTo(config('wallet.transaction.model', Transaction::class), 'deposit_id');
```
Comment (rez1dent3) [inline src/Models/Transfer.php:101]: not relevant. everything works without problems.

detail: https://github.com/bavix/laravel-wallet/pull/858#issuecomment-1906640926

File: src/Models/Transfer.php [hunk 2]
```diff
@@ -79,32 +80,32 @@ public function getTable(): string
...
     }
 
     /**
      * @return BelongsTo<Transaction, self>
      */
     public function withdraw(): BelongsTo
     {
-        return $this->belongsTo(Transaction::class, 'withdraw_id');
+        return $this->belongsTo(config('wallet.transaction.model', Transaction::class), 'withdraw_id');
```
Comment (rez1dent3) [inline src/Models/Transfer.php:109]: not relevant. everything works without problems.

detail: https://github.com/bavix/laravel-wallet/pull/858#issuecomment-1906640926

**Comment by gkmk** [comment]
@rez1dent3 i see. I think we should check the migrations then because i switched to **morph to** relationship just because of the database structure. 
About the **belongs to** relationships i find the definition on the model better just because it is not as magical as container binding and the responsibility is kept in one place. I also noticed that this is the pattern you have used inside other models. So it just keeps consistency.

**Comment by rez1dent3** [comment]
@gkmk okey. revert the changes with morphTo and I will merge.

**Pull request merged**